### PR TITLE
binance: trailing stop orders don't require stopPrice.

### PR DESCRIPTION
**Pull request by blair**
If a `trailingDelta` is given then the API docs say

> Unlike regular contingent orders, the `stopPrice` parameter is optional for trailing stop orders. If it is provided then the order will only start tracking price changes after the `stopPrice` condition is met. If the `stopPrice` parameter is omitted then the order starts tracking price changes from the next trade.

https://github.com/binance/binance-spot-api-docs/blob/master/faqs/trailing-stop-faq.md#how-do-i-place-a-trailing-stop-order

This is my best attempt at keeping the current way of indicating a stop limit order when `stopPrice` is not provided.

Feedback welcome on a better way of doing this.

### CODE REVIEW

**Comment by carlosmiei** [comment]
Hello @blair, thanks for your contribution, will check it soon

File: ts/src/binance.ts [hunk 1]
```diff
@@ -4200,8 +4200,9 @@ export default class binance extends Exchange {
         const triggerPrice = this.safeValue2 (params, 'triggerPrice', 'stopPrice');
         const stopLossPrice = this.safeValue (params, 'stopLossPrice', triggerPrice);  // fallback to stopLoss
         const takeProfitPrice = this.safeValue (params, 'takeProfitPrice');
-        const isStopLoss = stopLossPrice !== undefined;
-        const isTakeProfit = takeProfitPrice !== undefined;
+        const trailingDelta = this.safeValue (params, 'trailingDelta');
```
Comment (carlosmiei) [inline ts/src/binance.ts:4203]: @blair I'm not sure what this edit intends to fix, can you give an example?

For instance I tested with `binance createOrder "LTC/USDT" STOP_LOSS_LIMIT buy 10 20 '{"trailingDelta":10}'` and seems to be working

Comment (carlosmiei) [reply ts/src/binance.ts:4203]: @blair I understand and agree, I just think we should default to `isStopLoss` only (and not `istakeProfit`) for consistency's sake. I will update that and improve the docs

**Comment by carlosmiei** [comment]
LGTM 

```
p binance createOrder "LTC/USDT" limit buy 0.2 50                        
Python v3.10.9
CCXT v3.1.3
binance.createOrder(LTC/USDT,limit,buy,0.2,50)
{'amount': 0.2,
 'average': None,
 'clientOrderId': 'x-R4BD3S82c58a8a620aff7820421b87',
 'cost': 0.0,
 'datetime': '2023-05-22T17:43:35.335Z',
 'fee': {'cost': None, 'currency': None, 'rate': None},
 'fees': [{'cost': None, 'currency': None, 'rate': None}],
 'filled': 0.0,
 'id': '3509097955',
 'info': {'clientOrderId': 'x-R4BD3S82c58a8a620aff7820421b87',
          'cummulativeQuoteQty': '0.00000000',
          'executedQty': '0.00000000',
          'fills': [],
          'orderId': '3509097955',
          'orderListId': '-1',
          'origQty': '0.20000000',
          'price': '50.00000000',
          'selfTradePreventionMode': 'NONE',
          'side': 'BUY',
          'status': 'NEW',
          'symbol': 'LTCUSDT',
          'timeInForce': 'GTC',
          'transactTime': '1684777415335',
          'type': 'LIMIT',
          'workingTime': '1684777415335'},
 'lastTradeTimestamp': 1684777415335,
 'postOnly': False,
 'price': 50.0,
 'reduceOnly': None,
 'remaining': 0.2,
 'side': 'buy',
 'status': 'open',
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'timeInForce': 'GTC',
 'timestamp': 1684777415335,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```

```
p binance createOrder "LTC/USDT" "limit" buy 0.2 50 '{"trailingDelta":10}'

Python v3.10.9
CCXT v3.1.3
binance.createOrder(LTC/USDT,limit,buy,0.2,50,{'trailingDelta': 10})
{'amount': 0.2,
 'average': None,
 'clientOrderId': 'x-R4BD3S82ed93bdbad8ff26bcd37369',
 'cost': 0.0,
 'datetime': None,
 'fee': {'cost': None, 'currency': None, 'rate': None},
 'fees': [{'cost': None, 'currency': None, 'rate': None}],
 'filled': 0.0,
 'id': '3509098666',
 'info': {'clientOrderId': 'x-R4BD3S82ed93bdbad8ff26bcd37369',
          'cummulativeQuoteQty': '0.00000000',
          'executedQty': '0.00000000',
          'fills': [],
          'orderId': '3509098666',
          'orderListId': '-1',
          'origQty': '0.20000000',
          'price': '50.00000000',
          'selfTradePreventionMode': 'NONE',
          'side': 'BUY',
          'status': 'NEW',
          'symbol': 'LTCUSDT',
          'timeInForce': 'GTC',
          'trailingDelta': '10',
          'trailingTime': '1684777464442',
          'transactTime': '1684777464442',
          'type': 'STOP_LOSS_LIMIT',
          'workingTime': '-1'},
 'lastTradeTimestamp': 1684777464442,
 'postOnly': False,
 'price': 50.0,
 'reduceOnly': None,
 'remaining': 0.2,
 'side': 'buy',
 'status': 'open',
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'timeInForce': 'GTC',
 'timestamp': -1,
 'trades': [],
 'triggerPrice': None,
 'type': 'stop_loss_limit'}
```

```
 p binanceusdm createOrder "SOL/USDT:USDT" limit buy 1 15 --sandbox 
Python v3.10.9
CCXT v3.1.3
binanceusdm.createOrder(SOL/USDT:USDT,limit,buy,1,15)
{'amount': 1.0,
 'average': None,
 'clientOrderId': 'PvoK2aJmRF2lOTbK9ExrEA',
 'cost': 0.0,
 'datetime': '2023-05-22T17:45:06.477Z',
 'fee': {'cost': None, 'currency': None, 'rate': None},
 'fees': [{'cost': None, 'currency': None, 'rate': None}],
 'filled': 0.0,
 'id': '52751065',
 'info': {'avgPrice': '0.00',
          'clientOrderId': 'PvoK2aJmRF2lOTbK9ExrEA',
          'closePosition': False,
          'cumQty': '0',
          'cumQuote': '0.0000',
          'executedQty': '0',
          'orderId': '52751065',
          'origQty': '1',
          'origType': 'LIMIT',
          'positionSide': 'BOTH',
          'price': '15.0000',
          'priceProtect': False,
          'reduceOnly': False,
          'side': 'BUY',
          'status': 'NEW',
          'stopPrice': '0.0000',
          'symbol': 'SOLUSDT',
          'timeInForce': 'GTC',
          'type': 'LIMIT',
          'updateTime': '1684777506477',
          'workingType': 'CONTRACT_PRICE'},
 'lastTradeTimestamp': None,
 'postOnly': False,
 'price': 15.0,
 'reduceOnly': False,
 'remaining': 1.0,
 'side': 'buy',
 'status': 'open',
 'stopPrice': None,
 'symbol': 'SOL/USDT:USDT',
 'timeInForce': 'GTC',
 'timestamp': 1684777506477,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```

```
p binanceusdm createOrder "SOL/USDT:USDT" market buy 1  --sandbox
Python v3.10.9
CCXT v3.1.3
binanceusdm.createOrder(SOL/USDT:USDT,market,buy,1)
{'amount': 1.0,
 'average': 19.95,
 'clientOrderId': 'Tgxxks8AvoOfN8NJva5G7p',
 'cost': 19.95,
 'datetime': None,
 'fee': {'cost': None, 'currency': None, 'rate': None},
 'fees': [{'cost': None, 'currency': None, 'rate': None}],
 'filled': 1.0,
 'id': '52751066',
 'info': {'avgPrice': '19.9500',
          'clientOrderId': 'Tgxxks8AvoOfN8NJva5G7p',
          'closePosition': False,
          'cumQty': '1',
          'cumQuote': '19.9500',
          'executedQty': '1',
          'orderId': '52751066',
          'origQty': '1',
          'origType': 'MARKET',
          'positionSide': 'BOTH',
          'price': '0.0000',
          'priceProtect': False,
          'reduceOnly': False,
          'side': 'BUY',
          'status': 'FILLED',
          'stopPrice': '0.0000',
          'symbol': 'SOLUSDT',
          'timeInForce': 'GTC',
          'type': 'MARKET',
          'updateTime': '1684777524661',
          'workingType': 'CONTRACT_PRICE'},
 'lastTradeTimestamp': None,
 'postOnly': False,
 'price': 19.95,
 'reduceOnly': False,
 'remaining': 0.0,
 'side': 'buy',
 'status': 'closed',
 'stopPrice': None,
 'symbol': 'SOL/USDT:USDT',
 'timeInForce': 'GTC',
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'market'}
```

```
 p binance createOrder "LTC/USDT" market buy 0.2                          
Python v3.10.9
CCXT v3.1.3
binance.createOrder(LTC/USDT,market,buy,0.2)
{'amount': 0.2,
 'average': 91.01,
 'clientOrderId': 'x-R4BD3S825e657ba0b19846405c0c61',
 'cost': 18.202,
 'datetime': '2023-05-22T17:45:45.718Z',
 'fee': {'cost': None, 'currency': None, 'rate': None},
 'fees': [{'cost': 0.0002, 'currency': 'LTC'}],
 'filled': 0.2,
 'id': '3509099664',
 'info': {'clientOrderId': 'x-R4BD3S825e657ba0b19846405c0c61',
          'cummulativeQuoteQty': '18.20200000',
          'executedQty': '0.20000000',
          'fills': [{'commission': '0.00020000',
                     'commissionAsset': 'LTC',
                     'price': '91.01000000',
                     'qty': '0.20000000',
                     'tradeId': '289220890'}],
          'orderId': '3509099664',
          'orderListId': '-1',
          'origQty': '0.20000000',
          'price': '0.00000000',
          'selfTradePreventionMode': 'NONE',
          'side': 'BUY',
          'status': 'FILLED',
          'symbol': 'LTCUSDT',
          'timeInForce': 'GTC',
          'transactTime': '1684777545718',
          'type': 'MARKET',
          'workingTime': '1684777545718'},
 'lastTradeTimestamp': 1684777545718,
 'postOnly': False,
 'price': 91.01,
 'reduceOnly': None,
 'remaining': 0.0,
 'side': 'buy',
 'status': 'closed',
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'timeInForce': 'GTC',
 'timestamp': 1684777545718,
 'trades': [{'amount': 0.2,
             'cost': 18.202,
             'datetime': None,
             'fee': {'cost': 0.0002, 'currency': 'LTC'},
             'fees': [{'cost': 0.0002, 'currency': 'LTC'}],
             'id': '289220890',
             'info': {'commission': '0.00020000',
                      'commissionAsset': 'LTC',
                      'price': '91.01000000',
                      'qty': '0.20000000',
                      'tradeId': '289220890'},
             'order': None,
             'price': 91.01,
             'side': None,
             'symbol': 'LTC/USDT',
             'takerOrMaker': None,
             'timestamp': None,
             'type': None}],
 'triggerPrice': None,
 'type': 'market'}
```

**Review by carlosmiei — APPROVED** [review]

**Comment by blair** [comment]
Thanks for the quick merge!

**Pull request merged**